Fix stub clock driver and enable little cluster idle power state

Leo-Yan opened the pull request:
This patch series fix stub clock driver so let CPU cluster clocks can register with clock id, finally this will be used by cpufreq-dt to access CPU clocks. And based on latest released MCU firmware, the second patch is to enable little cluster idle power state so now have no hang issue.

Commit:
clk: hisilicon: hi3660: correct stub clock id

In the old code base it uses clock name to access clock nodes. When
change to use cpufreq-dt driver, we need support to use clock Id to
access these clock nodes, but current stub clock driver wrongly to
assign clock id and finally GPU clock overwrites other clocks, so
finally cpufreq cannot work.

This patch is to fix this issue to register clock nodes properly with
correct clock id.

Signed-off-by: Leo Yan <leo.yan@linaro.org>

Commit:
dts: hi3660: enable cluster0 idle state

Hisilicon releases new MCU firmware, which can resolve hang issue after
enable cluster level low power state. This dependency patch is included
in the repo: github.com/96boards-hikey/tools-images-hikey960; the commit
id is 9891e5c9ed9a ("fastboot.img: lpm3: sec_xloader: updated to fix
graphics performance issue"). So make sure the Hisilicon firmware has
been updated to latest version.

Signed-off-by: Leo Yan <leo.yan@linaro.org>

docularxu commented:
2017年3月22日 上午11:37，"Leo Yan" <notifications@github.com>写道：

This patch series fix stub clock driver so let CPU cluster clocks can
register with clock id, finally this will be used by cpufreq-dt to access
CPU clocks. And based on latest released MCU firmware,

Hi, Leo

would you please clarify where I can find "the latest released mcu
firmware"? is it submitted to github? a commit id maybe.

thank you


the second patch is to enable little cluster idle power state so now have
no hang issue.
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/96boards-hikey/linux/pull/31
Commit Summary

   - clk: hisilicon: hi3660: correct stub clock id
   - dts: hi3660: enable cluster0 idle state

File Changes

   - *M* arch/arm64/boot/dts/hisilicon/hi3660.dtsi
   <https://github.com/96boards-hikey/linux/pull/31/files#diff-0> (8)
   - *M* drivers/clk/hisilicon/clk-hi3660-stub.c
   <https://github.com/96boards-hikey/linux/pull/31/files#diff-1> (47)
   - *M* include/dt-bindings/clock/hi3660-clock.h
   <https://github.com/96boards-hikey/linux/pull/31/files#diff-2> (1)

Patch Links:

   - https://github.com/96boards-hikey/linux/pull/31.patch
   - https://github.com/96boards-hikey/linux/pull/31.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/96boards-hikey/linux/pull/31>, or mute the thread
<https://github.com/notifications/unsubscribe-auth/AHnKp2r319Vnxs1Sac-sJfvDR88-eu2eks5roJdngaJpZM4Mkr9p>
.

Leo-Yan commented:
On 22 March 2017 at 13:40, Guodong Xu <notifications@github.com> wrote:

> 2017年3月22日 上午11:37，"Leo Yan" <notifications@github.com>写道：
>
> This patch series fix stub clock driver so let CPU cluster clocks can
> register with clock id, finally this will be used by cpufreq-dt to access
> CPU clocks. And based on latest released MCU firmware,
>
> Hi, Leo
>
> would you please clarify where I can find "the latest released mcu
> firmware"? is it submitted to github? a commit id maybe.
>

I have added detailed info in second patche's commit log:

dts: hi3660: enable cluster0 idle state

Hisilicon releases new MCU firmware, which can resolve hang issue after
enable cluster level low power state. This dependency patch is included
in the repo: github.com/96boards-hikey/tools-images-hikey960; the commit
id is 9891e5c9ed9a ("fastboot.img: lpm3: sec_xloader: updated to fix
graphics performance issue"). So make sure the Hisilicon firmware has
been updated to latest version.

docularxu commented:
2017年3月22日 下午2:00，"Leo Yan" <notifications@github.com>写道：

On 22 March 2017 at 13:40, Guodong Xu <notifications@github.com> wrote:

> 2017年3月22日 上午11:37，"Leo Yan" <notifications@github.com>写道：
>
> This patch series fix stub clock driver so let CPU cluster clocks can
> register with clock id, finally this will be used by cpufreq-dt to access
> CPU clocks. And based on latest released MCU firmware,
>
> Hi, Leo
>
> would you please clarify where I can find "the latest released mcu
> firmware"? is it submitted to github? a commit id maybe.
>

I have added detailed info in second patche's commit log:



I see now.:)

sorry not checking earlier.





dts: hi3660: enable cluster0 idle state

Hisilicon releases new MCU firmware, which can resolve hang issue after
enable cluster level low power state. This dependency patch is included
in the repo: github.com/96boards-hikey/tools-images-hikey960; the commit
id is 9891e5c9ed9a ("fastboot.img: lpm3: sec_xloader: updated to fix
graphics performance issue"). So make sure the Hisilicon firmware has
been updated to latest version.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/96boards-hikey/linux/pull/31#issuecomment-288306502>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AHnKp-vegnll6lKbKFGxoCndXum8AJExks5roLj_gaJpZM4Mkr9p>
.
